### PR TITLE
Add semantic versioning and drafts to releases

### PR DESCRIPTION
**Pull request by tabgok**
* Uses auto-calculated semantic versioning
* Creates 'drafts' and 'releases' subdirectories under schema
* Moves generation to its own folder (for later partitioning)
* Consolidates requirements into requirements.txt
* README updates to reflect changes in usage

### CODE REVIEW

**Review by rarguelloF — COMMENTED** [review]
LGTM, nice improvements!

The only comment is that I would probably prefer to have the python script and anything necessary to execute it in the root folder (under `./semantic-core`) and anything that is generated under a `gen/<whatever>` folder, but this is just a personal opinion 🙂

File: semantic-core/generation/gen_semantic_defs.py [hunk 1]
```diff
@@ -553,23 +562,108 @@ def generate_schema(payload_type, version):
...
-    logger = logging.getLogger(__name__)
-
+    
     parser = argparse.ArgumentParser(description="Generate schema JSON file.")
-    parser.add_argument("version", type=str, help="Specify the version")
+    subparsers = parser.add_subparsers(dest='command')
 
+    # Create the parser for the "generate" command
+    generate_parser = subparsers.add_parser('generate', help="Generate something")
+    generate_parser.add_argument("--release", action="store_true", help="Create a new release version")
```
Comment (ajgajg1134) [inline semantic-core/generation/gen_semantic_defs.py:573]: what does action "store_true" mean here?

**Review by ajgajg1134 — APPROVED** [review]
Looks good to me! Just had one question about a string I didn't understand

File: semantic-core/generation/gen_semantic_defs.py (hunk 1 above)
Comment (tabgok) [reply semantic-core/generation/gen_semantic_defs.py:573]: `args.release` is always defined.  When specified, the value is `True`, when not specified the value is `False`.  

Its a little bit of syntactic sugar to make sure the field always exists and to specify a default.  `store_false` would have the opposite default behavior.

`command --release` =>  args.release is True

`command` => args.release is False

**Pull request merged**